src/storage/hummock_test/src/bin/replay/main.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![feature(bound_map)]
+#![feature(generators)]
+#![feature(stmt_expr_attributes)]
+#![feature(proc_macro_hygiene)]
+
+#[macro_use]
+mod replay_impl;
+
+use std::fs::File;
+use std::io::BufReader;
+use std::path::Path;
+use std::sync::Arc;
+
+use clap::Parser;
+use replay_impl::{get_replay_notification_client, GlobalReplayInterface};
+use risingwave_common::config::{
+    extract_storage_memory_config, load_config, StorageConfig, NO_OVERRIDE,
+};
+use risingwave_common::system_param::reader::SystemParamsReader;
+use risingwave_hummock_trace::{
+    GlobalReplay, HummockReplay, Operation, Record, Result, TraceReader, TraceReaderImpl, USE_TRACE,
+};
+use risingwave_meta::hummock::test_utils::setup_compute_env;
+use risingwave_meta::hummock::MockHummockMetaClient;
+use risingwave_object_store::object::parse_remote_object_store;
+use risingwave_storage::filter_key_extractor::{
+    FakeRemoteTableAccessor, FilterKeyExtractorManager,
+};
+use risingwave_storage::hummock::{HummockStorage, SstableStore, TieredCache};
+use risingwave_storage::monitor::{CompactorMetrics, HummockStateStoreMetrics, ObjectStoreMetrics};
+use risingwave_storage::opts::StorageOpts;
+use serde::{Deserialize, Serialize};
+
+const SST_OFFSET: u64 = 2147383647;
```

Comment:
Is this offset aimed to avoid overwriting existing SST? If so, it's not 100% safe.

Reply:
Yes. Is there any way to avoid this?